keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,397 @@
+<!-- toc -->
+- [cAdvisor-less, CRI-full Container and Pod Stats](#cadvisor-less-cri-full-container-and-pod-stats)
+  - [Summary](#summary)
+  - [Current State](#current-state)
+    - [Current Fulfiller of Summary API Stats &amp; Future Proposal](#current-fulfiller-of-summary-api-stats--future-proposal)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+      - [CRI Implementation](#cri-implementation)
+        - [ContainerStats additions](#containerstats-additions)
+        - [PodStats CRI additions](#podstats-cri-additions)
+      - [Kubelet](#kubelet)
+      - [cAdvisor](#cadvisor)
+    - [User Stories [optional]](#user-stories-optional)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks [optional]](#drawbacks-optional)
+  - [Alternatives [optional]](#alternatives-optional)
+    - [Notes [optional]](#notes-optional)
+<!-- /toc -->
+
+# cAdvisor-less, CRI-full Container and Pod Stats
+
+## Summary
+
+Kubelet relies on the [summary API](https://github.com/kubernetes/kubernetes/blob/release-1.20/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go) to provide node level stats, pod level stats and container level stats. 
```

Comment:
I think we were moving to the direction that Kubelet only provides the core metrics https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/core-metrics-pipeline.md#proposed-core-metrics.

And the CRI metrics set is defined based on that.

This seems to be against that direction.

Reply:
For context/reference, see https://github.com/kubernetes/kubernetes/issues/68522

Reply:
from [this](https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-724928827) and [this](https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-666636130) comment, it doesn't really seem like deprecating stats summary api is on the radar anymore (too many entities use them).

Reply:
I am adding a section that describes how we do not intend on dropping the stats summary api, and how it fits with the proposal described [here](https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-724928827)

Reply:
Thanks for adding the context!

Reply:
I have added an entry in Non-Goal that states we are not trying to fundamentally redesign Summary API (that can be done later)